legacy/reqcounter/reqcounter_test.go

```diff
@@ -0,0 +1,191 @@
+}
+
+func TestLog(t *testing.T) {
+	// Create multiple counters with unique logging intervals. Specify each counter's requests to match the unique interval time.
+	// By differing each counter's requests, we can identify which counter was logged.
+	millisecondCounter := NewRequestCounter(uint64(time.Millisecond))
+	millisecondCounter.Interval = time.Millisecond
+	secondCounter := NewRequestCounter(uint64(time.Second))
+	secondCounter.Interval = time.Second
+	minuteCounter := NewRequestCounter(uint64(time.Minute))
+	minuteCounter.Interval = time.Minute
+	hourCounter := NewRequestCounter(uint64(time.Hour))
+	hourCounter.Interval = time.Hour
+	// Create a new network monitor which holds all counters.
+	netMonitor := &rc.NetworkMonitor{
+		Counters: []*rc.RequestCounter{
+			&millisecondCounter,
+			&secondCounter,
+			&minuteCounter,
+			&hourCounter,
+		},
+	}
+	// Only these counter intervals should be logged during the test. Since we are measuring timings, the order
+	// in which they arrive matter. That is, shorter intervals are expected to log before longer intervals.
+	expects := []time.Duration{time.Millisecond, time.Second}
+	// Collect the counter intervals that were logged, in the order that they arrive.
+	collection := []time.Duration{}
+	// Since rc.Debug will be called concurrently, we must use a lock to protect against race conditions.
+	mutex := sync.Mutex{}
+	// Mock the logrus.Debug function, capturing which counters were logged.
+	rc.Debug = func(args ...interface{}) {
+		// Convert the args to a string for easier parsing.
+		s := fmt.Sprint(args[0])
+		// Determine which counter was retrieved.
+		// NOTE: Logs may manipulate the format of representing the interval, in terms of minutes, seconds, days...
+		// Therefore it is more robust to identify the counter by it's number of requests, since each is unique.
+		var interval time.Duration
+		if strings.Contains(s, fmt.Sprintf("%d requests", uint64(time.Millisecond))) {
+			interval = time.Millisecond
+		} else if strings.Contains(s, fmt.Sprintf("%d requests", uint64(time.Second))) {
+			interval = time.Second
+		} else if strings.Contains(s, fmt.Sprintf("%d requests", uint64(time.Minute))) {
+			interval = time.Minute
+		} else if strings.Contains(s, fmt.Sprintf("%d requests", uint64(time.Hour))) {
+			interval = time.Hour
+		} else {
+			// Resource counters are flushed after every log, thus logging "0 requests" every subsequent interval.
+			// Encountering this type of log is therefore not a problem, as is expected behavior.
+			return
+		}
+		// Ensure race conditions are avoided when manipulating the collection.
+		mutex.Lock()
+		defer mutex.Unlock()
+		// Add the interval to represent the counter to the collection.
+		collection = append(collection, interval)
+	}
+	// Begin logging each counter.
+	netMonitor.Log()
+	// Allow enough time to ensure ONLY the expected counters can log themselves.
+	time.Sleep(time.Second * 2)
```

Comment:
Again, please try to use a fake Sleep function. This is like dependency injection.

Reply:
Maybe having a fake Sleep can be part of a later design. It might complicate this area too much for this initial PR. Can you record this thought somewhere?

Reply:
Added "Test the Log function with fake sleep function" to issue: #358
This will be explored in another PR.